src/drivers/drivers.jl

```diff
@@ -89,8 +91,38 @@ function __init__()
     HDF5.HAS_PARALLEL[] = API._has_symbol(:H5Pset_fapl_mpio)
     HDF5.HAS_ROS3[] = API._has_symbol(:H5Pset_fapl_ros3)
 
-    @require MPI = "da04e1cc-30fd-572f-bb4f-1f8673147195" include("mpio.jl")
+    @static if !isdefined(Base, :get_extension)
+        @require MPI = "da04e1cc-30fd-572f-bb4f-1f8673147195" include("../../ext/MPIExt.jl")
+    end
 end
 
+# The docstring for `MPIO` basically belongs to the struct `MPIO` in
+# ext/MPIExt/MPIExt.jl. However, we need to document it here to make it
+# work smoothly both for Julia without package extensions (up to v1.8) and
+# with package extensions (v1.9 and newer).
+@doc """
+    MPIO(comm::MPI.Comm, info::MPI.Info)
+    MPIO(comm::MPI.Comm; kwargs....)
+
+The parallel MPI file driver. This requires the use of
+[MPI.jl](https://github.com/JuliaParallel/MPI.jl), and a custom HDF5 binary that has been
+built with MPI support.
+
+- `comm` is the communicator over which the file will be opened.
+- `info`/`kwargs` are MPI-IO options, and are passed to `MPI_FILE_OPEN`.
+
+# See also
+
+- [`HDF5.has_parallel`](@ref)
+- [Parallel HDF5](@ref)
+
+# External links
+
+- $(h5doc("H5P_SET_FAPL_MPIO"))
+- [Parallel HDF5](https://portal.hdfgroup.org/display/HDF5/Parallel+HDF5)
+"""
+function MPIO end
```

Comment:
Does this work? Or should it just be `MPIO`?

Reply:
I'm not sure I fully understand your question. The idea behind package extensions is that one can add new methods/specializations of functions available in the main package. That's why I set up this package extension so that we can keep using the same syntax as before but extend a function from HDF5.jl. That's related to my comment https://github.com/JuliaIO/HDF5.jl/pull/1062#issuecomment-1517984240 above.
If you have another suggestion how to handle this, please let me know.